### PR TITLE
Allow custom proxy host header to be set

### PR DESCRIPTION
**Pull request by MrJmpl3**
## Type of Change

- **Build Scripts:** 
- **Tool Source:**  Vue -> Proxy category
- **Something else:** 

## What issue does this relate to?

Trying to fix: #382 

### What should this PR do?

Move the proxy_set_header Host to per domain config

### What are the acceptance criteria?

Well, missing translations with the key 'proxyHeaderOverride', also in Spanish, I think the translation is 'Reemplazo de cabecera del proxy' or only 'Reemplazo de cabecera' because 'del proxy' is redudant.

In another languages, I don't know 😌

If all is correct, please add the tag 'hacktoberfest-accepted', Thanks

### CODE REVIEW

**Comment by MrJmpl3** [comment]
Oh, the test failed because the missing translation, and now? It's 'Google translate' time?

**Comment by MattIPv4** [comment]
@MrJmpl3 You can leave the strings in English in each of the i18n files and add `// TODO: translate` comments to each line (you'll find a few of these already in the i18n files)

File: src/nginxconfig/i18n/en/templates/domain_sections/reverse_proxy.js [hunk 1]
```diff
@@ -31,4 +31,5 @@ export default {
     reverseProxyCannotBeEnabledWithPhp: `${common.reverseProxy} cannot be enabled whilst ${common.php} is enabled.`,
     reverseProxyCannotBeEnabledWithPython: `${common.reverseProxy} cannot be enabled whilst ${common.python} is enabled.`,
     enableReverseProxy: `${common.enable} ${common.reverseProxyLower}`,
+    proxyHeaderOverride: 'Proxy header override',
```
Comment (MattIPv4) [inline src/nginxconfig/i18n/en/templates/domain_sections/reverse_proxy.js:34]:
```suggestion
    proxyHeaderOverride: 'Proxy Host header',
```
I think we should call out the header this affects here, for clarity (it also isn't really an override given the default is `$host`).

**Review by MattIPv4 — CHANGES_REQUESTED** [review]
:wave: Could you make sure the copyright year is bumped in the header of any file you've touched?

**Comment by MrJmpl3** [comment]
> 👋 Could you make sure the copyright year is bumped in the header of any file you've touched?

Ok, I update the year of copyright

**Review by MattIPv4 — CHANGES_REQUESTED** [review]
👀 Two more things, sorry!

- Could apply the `'Proxy header override'` ->  `'Proxy Host header'` change to all the i18n files?
- Can you change the key we're using in all the logic from being `proxyHeaderOverride` to `proxyHostHeader`? We override quite a few headers, but this setting is specifically the host header

**Comment by MrJmpl3** [comment]
> 👀 Two more things, sorry!
> 
>     * Could apply the `'Proxy header override'` ->  `'Proxy Host header'` change to all the i18n files?
> 
>     * Can you change the key we're using in all the logic from being `proxyHeaderOverride` to `proxyHostHeader`? We override quite a few headers, but this setting is specifically the host header

ok, wait some minutes

**Review by MattIPv4 — APPROVED** [review]
LGTM, tyvm

**Pull request merged**